pint/quantity.py

```diff
@@ -1879,12 +1879,22 @@ def __setitem__(self, key, value):
 
     def tolist(self):
         units = self._units
-        return [
-            self.__class__(value, units).tolist()
-            if isinstance(value, list)
-            else self.__class__(value, units)
-            for value in self._magnitude.tolist()
-        ]
+
+        try:
+            if not isinstance(self._magnitude.tolist(), Iterable):
+                value = self._magnitude.tolist()
+                return self.__class__(value, units)
+
+            return [
+                self.__class__(value, units).tolist()
+                if isinstance(value, list)
+                else self.__class__(value, units)
+                for value in self._magnitude.tolist()
+            ]
+        except AttributeError:
+            raise AttributeError(
+                f"Magnitude '{type(self._magnitude).__name__}' does not support tolist."
+            )
```

Comment:
Wouldn't it be much easier to just do
```suggestion
        values = self._magnitude.tolist()
        if not isinstance(values, list):
            values = [values]

        return [
            self.__class__(value, units).tolist()
            if isinstance(value, list)
            else self.__class__(value, units)
            for value in values
        ]
```
?

That way, we only call `tolist` once per dimension and the code is a little bit simpler

Reply:
Yeah thanks, I knew performance wise calling twice wasn't that good thx :). True I'll make the changes